src/pages/Home/Home.js

```diff
@@ -1,12 +1,166 @@
 import React from "react";
-import { Coins } from "components"
+import axios from "axios";
+import { default as _ } from "lodash";
+import queryString from "query-string";
+import { Coins, TableHeader } from "components";
+import { Container, StyledCol, StyledRow, StyledTitle } from "./Home.styles";
 
 export default class Home extends React.Component {
+  state = {
+    isLoading: false,
+    hasError: false,
+    coinList: [],
+    perPage: 10,
+    page: 1,
+    queryOrder: "market_cap_desc",
+    sortBy: "market_cap",
+    descending: true, // if true, sort in descending order, if false sort in ascending order
+  };
+
+  getCoins = async () => {
+    const { perPage, page, queryOrder } = this.state;
+    try {
+      this.setState({ isLoading: true });
+      const { data } = await axios(`
+      https://api.coingecko.com/api/v3/coins/markets?vs_currency=${this.props.currency}&order=${queryOrder}&per_page=${perPage}&page=${page}&price_change_percentage=1h%2C24h%2C7d&sparkline=true
+      `);
+      this.setState({ coinList: data, isLoading: false, hasError: false });
+    } catch (err) {
+      this.setState({ isLoading: false, hasError: err });
+    }
+  };
+
+  sortCoins = (item1, item2, key, descending) => {
+    if (item1[key] < item2[key]) return descending ? 1 : -1;
+    if (item1[key] > item2[key]) return descending ? -1 : 1;
+    return 0;
+  };
```

Comment:
sortCoins = (item1, item2) => {
    const {sortBy, descending} = this.state;

    if (item1[sortBy] < item2[sortBy]) return descending ? 1 : -1;
    if (item1[sortBy] > item2[sortBy]) return descending ? -1 : 1;
    return 0;
  };